components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -2520,3 +2520,30 @@ def object_table_query(request, objtype, objid, conn=None, **kwargs):
     tableData['parentId'] = ann['parentId']
     tableData['addedOn'] = ann['addedOn']
     return tableData
+
+
+@login_required()
+@jsonp
+def get_image_rdefs_json(request, img_id=None, conn=None, **kwargs):
+    """
+    Retrieves all rendering definitions for a given image (id).
+
+    Example:  /get_image_rdefs_json/1
+              Returns all rdefs for image with id 1
+
+    @param request:     http request.
+    @param img_id:      the id of the image in question
+    @param conn:        L{omero.gateway.BlitzGateway}
+    @param **kwargs:    unused
+    @return:            A dictionary with key 'rdefs' in the success case,
+                        one with key 'error' if something went wrong
+    """
+    try:
+        img = conn.getObject("Image", img_id)
+
+        if img is None:
+            return {'error': 'No image with id ' + str(img_id)}
+
+        return {'rdefs': img.getAllRenderingDefs()}
+    except:
+        return {'error': 'Failed to retrieve rdefs'}
```

Comment:
Our error handling in webgateway is not always consistent, but it could be handy to either log the error or return it to user as we do elsewhere, E.g:
```
    return HttpJavascriptResponseServerError('("error in call","%s")' % traceback.format_exc())
or 
    logger.debug(traceback.format_exc())
    return {'error': 'Failed to retrieve rdefs'}
```